benchmarks/src/jmh/java/io/grpc/benchmarks/netty/FlowControlledMessagesPerSecondBenchmark.java

```diff
@@ -0,0 +1,105 @@
+package io.grpc.benchmarks.netty;
+
+import org.openjdk.jmh.annotations.AuxCounters;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Benchmark measuring messages per second received from a streaming server. The server
+ * is obeying outbound flow-control.
+ */
+@State(Scope.Benchmark)
+@Fork(1)
+public class FlowControlledMessagesPerSecondBenchmark extends AbstractBenchmark {
+
+  @Param({"1", "2", "4"})
+  public int channelCount = 1;
+
+  @Param({"1", "2", "10", "100"})
+  public int maxConcurrentStreams = 1;
+
+  @Param
+  public ExecutorType clientExecutor = ExecutorType.DIRECT;
+
+  @Param({"SMALL"})
+  public PayloadSize responseSize = PayloadSize.SMALL;
+
+  private static AtomicLong callCounter;
+  private AtomicBoolean completed;
+
+  /**
+   * Use an AuxCounter so we can measure that calls as they occur without consuming CPU
+   * in the benchmark method.
+   */
+  @AuxCounters
+  @State(Scope.Thread)
+  public static class AdditionalCounters {
+
+    @Setup(Level.Iteration)
+    public void clean() {
+      callCounter.set(0);
+    }
+
+    public long messagesPerSecond() {
+      return callCounter.get();
+    }
+  }
+
+  /**
+   * Setup with direct executors, small payloads and the default flow-control window.
+   */
+  @Setup(Level.Trial)
+  public void setup() throws Exception {
+    super.setup(clientExecutor,
+        ExecutorType.DIRECT,
+        PayloadSize.SMALL,
+        responseSize,
+        FlowWindowSize.MEDIUM,
+        ChannelType.NIO,
+        maxConcurrentStreams,
+        channelCount);
+    callCounter = new AtomicLong();
+    completed = new AtomicBoolean();
+    startFlowControlledStreamingCalls(maxConcurrentStreams, callCounter, completed, 1);
+  }
+
+  /**
+   * Stop the running calls then stop the server and client channels.
+   */
+  @TearDown(Level.Trial)
+  public void stopChannelsAndServers() throws Exception {
```

Comment:
Oh I see ... I think you're missing `@Override`. Maybe rename the base method too?